corehq/ex-submodules/casexml/apps/phone/data_providers/case/livequery.py

```diff
@@ -58,6 +58,51 @@ def enliven(case_id):
             for next_id in chain(ext_ids, host_ids):
                 enliven(next_id)
 
+    def classify(index):
+        """Classify index as either live or extension with live status pending
+
+        This closuer mutates `next_ids`, `extensions_by_host`,
```

Comment:
That's true, if you're into class-oriented thinking, which we often do at Dimagi. In this case you can think of `do_livequery` sort of like a class, with its local variables being members that would be bound to `self` in a class. The inner closures `enliven` and `classify` could be thought of like private methods that mutate state.

I personally prefer the syntax of function closures to helper classes like the one you described because there is less boilerplate syntax involved: a `LiveQueryHelper` instance has no useful purpose other than to call it's `main()` method, so why not just make it a function?